code/controllers/subsystems/mapping.dm

```diff
@@ -68,3 +89,22 @@ SUBSYSTEM_DEF(mapping)
 			space_ruins_templates[MT.name] = MT
 		else if(istype(MT, /datum/map_template/ruin/away_site))
 			away_sites_templates[MT.name] = MT
+
+
+//This loops through every zlevel that exists in memory, and throws up errors about those that don't have datums setup
+//We do this by looping through numbers until we fail to retrieve a tile
+/hook/roundstart/proc/check_zlevel_datums()
+	var/done = FALSE
+	var/index = 1
+	while(!done)
+		var/turf/T = locate(1,1,index)
+		if (istype(T))//If we found T then this zlevel exists
+			var/datum/level/L = SSmapping.zlevels["[T.z]"]
+			if (!istype(L))
+				log_world("Error: Z level [T.z] has no associated level datum.")
```

Comment:
Debug message.

Reply:
this one is intentional, it should be left in permanantly.
It warns map makers that they've screwed up by not placing a level landmark